src/raft/raft.rs

```diff
@@ -725,6 +742,57 @@ impl<T: Storage> Raft<T> {
         *maybe_commit = true;
     }
 
+    fn handle_transfer_leader(&mut self, m: &Message) {
+        let lead_transferee = m.get_from();
+        if let Some(last_lead_transferee) = self.lead_transferee {
+            if last_lead_transferee == lead_transferee {
+                info!("{} [term {}] transfer leadership to {} is in progress, ignores request to \
+                       same node {}",
+                      self.id,
+                      self.term,
+                      lead_transferee,
+                      lead_transferee);
+                return;
+            }
+            self.abort_leader_transfer();
+            info!("{} [term {}] abort transfer leadership to {}",
+                  self.id,
+                  self.term,
+                  last_lead_transferee);
+        }
+        if lead_transferee == self.id {
+            if self.lead_transferee.is_none() {
```

Comment:
we did use a last_lead_transferee, several lines before here.
a little difference is etcd use lead_transferee uint64, equal to 0 means None. 
while I think use Option<u64> conform to rust Idioms.